### PR TITLE
Add RELEASE.txt entry for API tests

### PR DESCRIPTION
**Pull request by jhendersonHDF**

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -47,6 +47,48 @@ New Features
 
     Configuration:
     -------------
+    - Added new CMake options for building and running HDF5 API tests
+      (Experimental)
+
+      HDF5 API tests are an experimental feature primarily targeted at
+      HDF5 VOL connector authors that is currently being developed.
```
Comment (bmribler) [inline release_docs/RELEASE.txt]: I'm not sure about the part after "targeted at", could you look it over?

Comment (jhendersonHDF) [reply release_docs/RELEASE.txt]: Hmm, maybe some commas would help distinguish the parts of the sentence here?

Comment (lrknox) [reply release_docs/RELEASE.txt]: one possibility:  targeted at authors of HDF5 VOL connectors currently being developed.

Comment (jhendersonHDF) [reply release_docs/RELEASE.txt]: To me that reads as though it's talking about being targeted at connectors that are currently being developed. That's true, but this is meant to say that the tests feature is currently being develop. I guess more like

"HDF5 API tests are an experimental feature, primarily targeted at HDF5 VOL connector authors, that is currently being developed."

**Review by derobins — APPROVED** [review]

File: release_docs/RELEASE.txt (hunk 1 above)
Comment (bmribler) [reply release_docs/RELEASE.txt]: I have no idea whom exactly supposed to be targeted at.  Another possibility: "HDF5 API tests are an experimental feature, primarily targeted at authors, who are currently developing applications using HDF5 VOL connector"?

Comment (hyoklee) [reply release_docs/RELEASE.txt]: Rewrite "HDF5 API tests are a work-in-progress experimental feature that is primarily aimed at HDF5 VOL connector authors."

**Pull request merged**